rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java

```diff
@@ -0,0 +1,741 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static co.rsk.peg.PegTestUtils.createFederation;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class PegUtilsTest {
+    private static final BridgeConstants bridgeMainnetConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcMainnetParams = bridgeMainnetConstants.getBtcParams();
+    private static final Context context = new Context(bridgeMainnetConstants.getBtcParams());
+    ActivationConfig.ForBlock activations = ActivationConfigsForTest.tbd600().forBlock(0);
+
+    private BridgeStorageProvider provider;
+
+    private static final int FIRST_OUTPUT_INDEX = 0;
+    private static final int FIRST_INPUT_INDEX = 0;
+    private Address unknownAddress;
+
+    @BeforeEach
+    void init() {
+        provider = mock(BridgeStorageProvider.class);
+        unknownAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "unknown");
+    }
+
+    @Test
+    void test_getTransactionType_before_tbd_600() {
+        // Arrange
+        ActivationConfig.ForBlock fingerrootActivations = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        Wallet liveFederationWallet = mock(Wallet.class);
+        BtcTransaction btcTransaction = mock(BtcTransaction.class);
+
+        // Act
+        IllegalStateException exception = Assertions.assertThrows(IllegalStateException.class, () -> {
+            PegUtils.getTransactionType(fingerrootActivations, provider, liveFederationWallet, btcTransaction);
+        });
+
+        // Assert
+        String expectedMessage = "Can't call this method before RSKIP379 activation.";
+        Assertions.assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    void test_getTransactionType_tx_sending_funds_to_unknown_address() {
+        // Arrange
+        Federation activeFederation = bridgeMainnetConstants.getGenesisFederation();
+        Wallet liveFederationWallet = new BridgeBtcWallet(context, Collections.singletonList(activeFederation));
+
+        Coin minimumPeginTxValue = bridgeMainnetConstants.getMinimumPeginTxValue(activations);
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTransaction.addOutput(minimumPeginTxValue, unknownAddress);
+
+        // Act
+        PegTxType pegTxType = PegUtils.getTransactionType(
+            activations,
+            provider,
+            liveFederationWallet,
+            btcTransaction
+        );
+
+        // Assert
+        Assertions.assertEquals(PegTxType.UNKNOWN, pegTxType);
+    }
+
+    @Test
+    void test_getTransactionType_pegin_below_minimum_active_fed() {
```

Comment:
Do We Need The Pegin/Pegout Amount To Determine The Transaction Type?

Reply:
The legacy method does, but since this is one that only aims to identify the type without checking if tx is valid, I wanted to assert the amount is not being used anymore to check pegins.